pyccel/ast/builtin_objects/list_functions.py

```diff
@@ -0,0 +1,80 @@
+# coding: utf-8
+#------------------------------------------------------------------------------------------#
+# This file is part of Pyccel which is released under MIT License. See the LICENSE file or #
+# go to https://github.com/pyccel/pyccel/blob/master/LICENSE for full license details.     #
+#------------------------------------------------------------------------------------------#
+"""
+The List container has a number of built-in methods that are 
+always available.
+
+This module contains objects which describe these methods within Pyccel's AST.
+"""
+
+from pyccel.ast.datatypes import NativeVoid, NativeGeneric, NativeHomogeneousList
+from pyccel.ast.internals import PyccelInternalFunction, get_final_precision
+
+
+class ListAppend(PyccelInternalFunction):
+    """
+    Represents a call to the .append() method.
+
+    Represents a call to the .append() method of an object with a list type,
+    which adds an element to the end of the list. This method returns `None`.
+    The append method is called as follows:
+
+    >>> a = [1]
+    >>> a.append(2)
+    >>> print(a)
+    [1, 2]
+
+    Parameters
+    ----------
+    lst_bound_arg : TypedAstNode
+        The variable representing the list.
+    
+    new_elem : TypedAstNode
+        The argument passed to append() method.
+    """
+    __slots__ = ("_lst_bound_arg", "_append_arg")
+    _dtype = NativeVoid()
+    _shape = None
+    _order = None
+    _rank = 0
+    _precision = -1
+    _class_type = NativeHomogeneousList()
+    name = 'append'
+
+    def __init__(self, lst_bound_arg, new_elem) -> None:
+        list_precision = get_final_precision(lst_bound_arg)
+        check = (getattr(lst_bound_arg, 'order', None) == getattr(new_elem, 'order', None))
+        conditions = (
+            new_elem.dtype is not NativeGeneric() and
+            lst_bound_arg.dtype == new_elem.dtype and
+            list_precision == get_final_precision(new_elem) and
```

Comment:
I imagine that you are using `get_final_precision` to match what is done in `builtins.py`. There we have been a little generous. Probably overly generous I suspect it would be better removed.

By using `get_final_precision` we allow the insertion of objects which would be homogeneous in C/Fortran. So you can then do:
```python
a = [1,2,3]
a.append(np.int64(4))
```
(or in Windows where the default precision is smaller
```python
a = [1,2,3]
a.append(np.int32(4))
```
)

Reply:
i see, what can i replace it with to check for the precision ?

Reply:
`lst_bound_arg.precision == new_elem.precision`